src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write code here
+const headTable = document.querySelector('thead');
+const bodyTable = document.querySelector('tbody');
+const rows = [...bodyTable.rows];
+
+headTable.addEventListener('click', e => {
+  if (!e.target.tagName === 'th') {
+    return;
+  };
+
+  const targetIndex = e.target.cellIndex;
+
+  function numberOptimizer(str) {
+    return str.replace(/[$,]/g, '');
+  };
+
+  const sortList = rows.sort((a, b) => {
+    const rowOne = numberOptimizer(a.cells[targetIndex].textContent);
+    const rowTwo = numberOptimizer(b.cells[targetIndex].textContent);
+
+    if (typeof rowOne === 'number') {
+      return rowOne - rowTwo;
```

Comment:
It still compares as `string`, because everything in `textContent` always is `string`. You should try to transform it into `number` and check if it's valid (`isNaN`)

![image](https://github.com/mate-academy/js_sort_table_DOM/assets/86923089/7edbdf86-1b94-4336-ac42-67c4094701f8)